extensions/2.0/Vendor/EXT_feature_metadata/README.md

```diff
@@ -0,0 +1,1691 @@
+
+Each buffer view `byteOffset` must be aligned to a multiple of 8 bytes. If the buffer view's buffer is the GLB-stored `BIN` chunk the byte offset is measured relative to the beginning of the GLB. Otherwise it is measured relative to the beginning of the buffer.
+
+### Feature Textures
+
+Feature textures (not to be confused with [Feature ID Textures](#feature-id-texture)) use textures rather than parallel arrays to store values. Feature textures are accessed directly by texture coordinates, rather than feature IDs. Feature textures are especially useful when texture mapping high frequency data to less detailed 3D surfaces.
+
+Feature textures use the [Raster Format](https://github.com/CesiumGS/3d-tiles/tree/3d-tiles-next/specification/Metadata#raster-format) of the [Cesium 3D Metadata Specification](https://github.com/CesiumGS/3d-tiles/tree/3d-tiles-next/specification/Metadata) with a few additional constraints:
+
+* A scalar property cannot be encoded into multiple channels. For example, it is not possible to encode a `UINT32` property in an `RGBA8` texture.
+* Components of fixed-length array properties must be separate channels within the same texture.
+* Variable-length arrays are not supported.
+
+Additionally, the data type and bit depth of the image must be compatible with the property type. An 8-bit per pixel RGB image is only compatible with `UINT8` or normalized `UINT8` properties, and array properties thereof with three components or less. Likewise, a floating point property requires a floating point-compatible image format like KTX2 which may require additional extensions.
+
+Feature textures are defined with the following steps:
+
+1. A class is defined in the root `EXT_feature_metadata` extension object. This is used to describe the metadata in the texture.
+2. A feature texture is defined in the root `EXT_feature_metadata.featureTextures` object. This must reference the class ID defined in step 1.
+3. A feature texture is associated with a primitive by listing the feature texture ID in the `primitive.EXT_feature_metadata.featureTextures` array.
+<img src="figures/feature-texture.png"  alt="Feature Texture" width="500">
+
+_Class and feature texture_
+
+```jsonc
+{
+  "extensions": {
+    "EXT_feature_metadata": {
+      "schema": {
+        "classes": {
+          "heatSample": {
+            "properties": {
+              "heatSample": {
+                "type": "UINT8",
+                "normalized": true
+              }
+            }
+          }
+        }
+      },
+      "featureTextures": {
+        "heatLossTexture": {
+          "class": "heatSample",
+          "properties": {
+            "heatLoss": {
+              "texture": {
+                "index": 0,
+                "texCoord": 0
+              },
+              "channels": "r"
+            }
+          }
+        }
```

Comment:
Conceptually, what does 1 feature texture represent? Should it map 1:1 to a specific class, a specific texture, all textures associated with a particular class, or something else? Based on the schema above, I'd assume this is valid...

```json
"featureTextures": {
    "texture1": {
        "class": "heatSample",
        "properties": {
            "heatLoss": {
                "texture": {
                    "index": 0,
                    "texCoord": 0
                },
                "channels": "r"
            },
            "heatGain": {
                "texture": {
                    "index": 1,
                    "texCoord": 0
                },
                "channels": "r"
            }
        }
    },
    "texture2": {
        "class": "heatSample",
        "properties": {
            "thermalInsulation": {
                "texture": {
                    "index": 0,
                    "texCoord": 0
                },
                "channels": "r"
            }
        }
    },
}
```
... containing one feature texture that consists of multiple textures, and two feature textures that describe the same class. Is that intended? I'm wondering if it's too much flexibility and should be restricted to tighter mapping.

Reply:
meeting notes: let's open PR and review implications

Reply:
I complete agree on the need to simplify feature textures.

They're an important use case but they feel _different_ than the rest of the extension.

Reply:
Perhaps something like this?

```
// ... root
"EXT_feature_metadata": {
  "schema": {
    "classes": {
      "tree": {
        "properties": {
          "height": {"type": "FLOAT32"},
          "birdCount": {"type": "UINT8"},
          "leafiness": {"type": "UINT8"}
        }
      }
    },
  },
  "featureTables": [{
    "class": "tree", // not indexed; schema may be external.
    "count": 10,
    "properties": {
      "height": { "bufferView": 0 },
      "birdCount": {"bufferView": 1 },
      "leafiness": {
        "texture": {
          "index": 0,
          "texCoord": 0,
          "channels": "r"
        }
      }
    }
  }],
}
```

Helpful parallelism between property data stored in buffers and in textures, and resolves a question I had about whether non-optional properties are required to be listed in the feature table if they can be sourced from a texture.

It does remove the flexibility of overriding individual feature textures at the primitive level, but that flexibility doesn't exist for feature property values in buffer views... so may not be necessary here? Different primitives can still have different UVs of course.

Reply:
@donmccurdy what would the corresponding primitive JSON look like?

I do like the symmetry of this JSON, though if feature textures are now contained in the feature tables, wondering how this will interact with the primitives, given that you access the texture via UVs, not feature ID

Reply:
Hm, given https://github.com/CesiumGS/glTF/pull/9#discussion_r701992644 this is probably not the right approach.

I think I'd been under the impression that both 1:1 feature-and-value associations (feature table), and 1:many associations (feature texture) would be defined in a single class, with some semantic meaning associated with the class. But practically, you probably do want separate classes for feature tables vs. feature textures, the two don't really even share the same concept of "a feature". Will think about this a bit more, but I'm leaning against putting textures into the `featureTables` list now.

Reply:
Let's consider this thread resolved — the separate table vs. raster formats are best separate, and the discussion of simplifying feature textures can continue on https://github.com/CesiumGS/glTF/pull/9#discussion_r701513015.